server/s3_object_data_copier.cc

```diff
@@ -362,6 +371,47 @@ void S3ObjectDataCopier::copy(
   s3_log(S3_LOG_DEBUG, "", "%s Exit", __func__);
 }
 
+void S3ObjectDataCopier::copy_part_fragment_in_single_source(
+    std::vector<struct s3_part_frag_context> fragment_context_list,
+    std::function<bool(void)> check_shutdown_and_rollback,
+    std::function<void(void)> on_success,
+    std::function<void(void)> on_failure) {
+  s3_log(S3_LOG_INFO, request_id, "%s Entry\n", __func__);
+  assert(check_shutdown_and_rollback);
+  assert(on_success);
+  assert(on_failure);
+  this->check_shutdown_and_rollback = std::move(check_shutdown_and_rollback);
+  this->on_success = std::move(on_success);
+  this->on_failure = std::move(on_failure);
+  part_fragment_context_list = fragment_context_list;
+  copy_parts_fragment_in_single_source = true;
+  set_next_part_context();
+  read_data_block();
+  s3_log(S3_LOG_INFO, request_id, "%s Exit\n", __func__);
+}
+
+void S3ObjectDataCopier::set_next_part_context() {
+  s3_log(S3_LOG_INFO, request_id, "%s Entry\n", __func__);
+  assert(non_zero(part_fragment_context_list[part_vector_index].motr_OID));
+  assert(part_fragment_context_list[part_vector_index].item_size > 0);
+  assert(part_fragment_context_list[part_vector_index].layout_id > 0);
+  motr_unit_size = S3MotrLayoutMap::get_instance()->get_unit_size_for_layout(
```

Comment:
Is `motr_unit_size` used anywhere in function or referenced somewhere else?

Reply:
referenced in read_data_block